service/health/searcher/types.go

```diff
@@ -72,3 +74,20 @@ func NewNodesStatus(cluster clusterInfo) []NodeStatus {
 	}
 	return result
 }
+
+func FindInAddresses(addresses []v1.NodeAddress, address_type v1.NodeAddressType) string {
+	for _, entry := range addresses {
+		if entry.Type == address_type {
+			return entry.Address
+		}
+	}
+	return "" // TODO: Handle behavior if node doesn't have a type
```

Comment:
Why can we not handle this here? And what do you think should happen?

Reply:
I don't see a reason for this to be an error since we're just exposing what we get from the cluster. So I think the empty string would be fine, unless there is a style rule or best practice or an error you think we should issue instead

Reply:
I was mostly referring to the TODO - I am fine with an empty string as this should _never_ be the case in any real scenario.

Reply:
It seems unlikely that we could get this far if the node doesn't have an internal IP address, but it could be possible for other types of addresses (ExternalIP for example). So empty string should be ok I think